staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -136,20 +136,26 @@ func TestGetServerGroupsWithTimeout(t *testing.T) {
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		// first we need to write headers, otherwise http client will complain about
 		// exceeding timeout awaiting headers, only after we can block the call
-		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Connection", "keep-alive")
+		if wf, ok := w.(http.Flusher); ok {
+			wf.Flush()
+		}
 		<-done
 	}))
 	defer server.Close()
 	defer close(done)
-	tmp := defaultTimeout
-	defaultTimeout = 2 * time.Second
```

Comment:
This needs to be here to ensure we are hitting that path. Without it this test will last definitely longer than it should be.

Reply:
Discussed this in depth with David on IRC, I'm ok with this change.